frontend/src/views/Collection/index.tsx

```diff
@@ -1,14 +1,114 @@
+import { Button, Classes, H3, H4, Intent, UL } from "@blueprintjs/core";
+import { IconNames } from "@blueprintjs/icons";
 import { RouteComponentProps } from "@reach/router";
+import psl from "psl";
 import React, { FC } from "react";
+import { Link, LINK_TYPE } from "src/common/entities";
+import { useCollection } from "src/common/queries/collections";
+import {
+  CenterAlignedDiv,
+  CollectionButtons,
+  CollectionInfo,
+  DatasetContainer,
+  Description,
+  LinkContainer,
+  StyledButton,
+  ViewGrid,
+} from "./style";
 
 interface RouteProps {
-  id: string;
+  id?: string;
 }
 
 export type Props = RouteComponentProps<RouteProps>;
 
+const getDomain = (url: string): string | null => {
+  let hostname;
+  if (url.indexOf("//") > -1) {
+    hostname = url.split("/")[2];
+  } else {
+    hostname = url.split("/")[0];
+  }
+  const parsedURL = psl.parse(hostname);
+  if (parsedURL.error) {
+    console.error(parsedURL.input, parsedURL.error);
+    return null;
+  }
+  return parsedURL.domain;
+};
+
+const RenderEmptyDatasets = () => {
+  return (
+    <CenterAlignedDiv>
+      <H4>No datasets uploaded</H4>
+      <div>
+        Before you begin uploading dataset files:
+        <UL>
+          <li>
+            You must validate your dataset locally. We provide a local CLI
+            script to do this. Learn More
+          </li>
+          <li>
+            We only support adding datasets in the h5ad format at this time.
+          </li>
+        </UL>
+      </div>
+      <Button
+        intent={Intent.PRIMARY}
+        outlined
+        text={"Add Dataset from Dropbox"}
+      />
+    </CenterAlignedDiv>
+  );
+};
+
+const renderLinks = (links: Link[]) => {
```

Comment:
<img width="1418" alt="Screen Shot 2020-11-04 at 11 53 28 AM" src="https://user-images.githubusercontent.com/6309723/98162733-a3fb1180-1e96-11eb-868f-2ad9a26ec43e.png">

The clickable area for the links extend to the right beyond the rendered text, can we shrink the clickable area to be just the rendered text please? Thank you!

Reply:
And the link type text seems to be the raw enum, I created type options in `frontend/src/components/CreateCollectionModal/components/Content/components/LinkInput/index.tsx:24`

```ts
export const OPTIONS = {
  [TYPES.DOI]: { text: "DOI", value: TYPES.DOI },
  [TYPES.RAW_DATA]: { text: "Raw Data", value: TYPES.RAW_DATA },
  [TYPES.PROTOCOL]: { text: "Protocol", value: TYPES.PROTOCOL },
  [TYPES.LAB_WEBSITE]: { text: "Lab Website", value: TYPES.LAB_WEBSITE },
  [TYPES.OTHER]: { text: "Other", value: TYPES.OTHER },
};
```

Maybe we can move that to `entities.ts` and reuse it in both Collection page and Create Collection component?

Reply:
@tihuan Oh nice, that sounds good to me! I'll put that in in just a bit

Reply:
Awesome thank you!